brushfire-core/src/main/scala/com/stripe/brushfire/Samplers.scala

```diff
@@ -2,18 +2,18 @@ package com.stripe.brushfire
 
 import com.twitter.algebird._
 
-object SingleTreeSampler extends Sampler[Any] {
+object SingleTreeSampler extends Sampler[Any, Any] {
   val numTrees = 1
-  def timesInTrainingSet(id: String, timestamp: Long, treeIndex: Int) = 1
-  def includeInValidationSet(id: String, timestamp: Long, treeIndex: Int) = true
-  def includeFeature(name: Any, treeIndex: Int, leafIndex: Int) = true
+  def timesInTrainingSet(metadata: Any, treeIndex: Int) = 1
+  def includeInValidationSet(metadata: Any, treeIndex: Int) = true
+  def includeFeature(metadata: Any, name: Any, treeIndex: Int, leafIndex: Int) = true
 }
 
-case class KFoldSampler(numTrees: Int) extends Sampler[Any] {
+case class KFoldSampler[-M](id: M => String, numTrees: Int) extends Sampler[M, Any] {
```

Comment:
I wonder if we want `id: M=>String` here or if it's better to have some kind of `M:Hashable` which builds in the murmur hashing. (Unfortunately Algebird has never really standardized a Hashable typeclass...)